substrate-node/pallets/pallet-tfgrid/src/grid_migration.rs

```diff
@@ -0,0 +1,263 @@
+        };
+
+        migrated_count += 1;
+
+        Some(new_entity)
+    });
+
+    info!(
+        " <<< Entity storage updated! Migrated {} Entities ✅",
+        migrated_count
+    );
+
+    // Return the weight consumed by the migration.
+    T::DbWeight::get().reads_writes(migrated_count as Weight + 1, migrated_count as Weight + 1)
+}
+
+fn migrate_nodes<T: Config>() -> frame_support::weights::Weight {
+    info!(" >>> Migrating nodes storage...");
+
+    let mut migrated_count = 0;
+
+    // We transform the storage values from the old into the new format.
+    Nodes::<T>::translate::<deprecated::Node<PubConfigOf<T>, InterfaceOf<T>>, _>(|k, node| {
+        info!("     Migrated node for {:?}...", k);
+
+        let location = match get_location::<T>(&node) {
+            Ok(loc) => loc,
+            Err(e) => {
+                info!(
+                    "location: [city: {:?}, country: {:?}, latitude: {:?}, longitude: {:?}]",
+                    core::str::from_utf8(&node.city).unwrap(),
+                    core::str::from_utf8(&node.country).unwrap(),
+                    core::str::from_utf8(&node.location.latitude).unwrap(),
+                    core::str::from_utf8(&node.location.longitude).unwrap()
+                );
+                info!("failed to parse location for node: {:?}, error: {:?}", k, e);
+                let default_location_input = LocationInput {
+                    city: BoundedVec::try_from(node::DEFAULT_CITY_NAME.to_vec()).unwrap(),
+                    country: BoundedVec::try_from(node::DEFAULT_COUNTRY_NAME.to_vec()).unwrap(),
+                    latitude: BoundedVec::try_from(node::DEFAULT_LATITUDE.to_vec()).unwrap(),
+                    longitude: BoundedVec::try_from(node::DEFAULT_LONGITUDE.to_vec()).unwrap(),
+                };
+                info!("set default location for node");
+                <T as Config>::Location::try_from(default_location_input).unwrap()
+            }
+        };
+
+        let serial_number_input: SerialNumberInput =
+            BoundedVec::try_from(node.serial_number.clone()).unwrap();
+        let serial_number = match <T as Config>::SerialNumber::try_from(serial_number_input) {
+            Ok(serial) => serial,
+            Err(e) => {
+                info!(
+                    "serial number: {:?}",
+                    core::str::from_utf8(&node.serial_number).unwrap()
+                );
+                info!(
+                    "failed to parse serial number for node: {:?}, error: {:?}",
+                    k, e
+                );
+                let default_serial_number_input: SerialNumberInput =
+                    BoundedVec::try_from(node::DEFAULT_SERIAL_NUMBER.to_vec()).unwrap();
+                info!("set default serial number for node");
+                <T as Config>::SerialNumber::try_from(default_serial_number_input).unwrap()
+            }
+        };
+
+        let new_node = TfgridNode::<T> {
+            version: 6, // ??
```

Comment:
which version here?
@DylanVerstraete

Reply:
Leave it like that, we will be deprecating this versioningen